src/conn/conn_compact.c

```diff
@@ -18,6 +21,61 @@ __compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_find_next_uri --
+ *     Given a URI, find the next one in the metadata file that is eligible for compaction.
+ */
+static int
+__background_compact_find_next_uri(WT_SESSION_IMPL *session, WT_ITEM *uri, WT_ITEM *next_urip)
```

Comment:
This can go back to `next_uri`. 

This is mentioned briefly on the C style [wiki page](https://wiki.corp.mongodb.com/pages/viewpage.action?pageId=91009507) (Search for "Names for function output parameters")

Basically if you're passing the address of a thing to a function in the expectation the function will fill it in, then we use the `p` suffix.

So if we expected this function to allocate or create a WT_ITEM and pass it back, then the caller would pass in a WT_ITEM ** and the variable name would end in `p`. But here we're just passing the WT_ITEM so the function can fill in the caller's WT_ITEM, so the `p`, isn't necessary.

Sorry. I feel like I'm giving you confusing and conflicting feedback here. It might be useful to look at existing examples in the code. `__wt_logrec_alloc`, for example allocates and returns a `WT_ITEM`, so the argument is `WT_ITEM **logrecp`. But `__wt_raw_to_hex` just writes some data into the `WT_ITEM` passed to it. So its argument is `WT_ITEM *to`.

Reply:
Thank you, @keitharnoldsmith, I really appreciate the details and the examples! I have updated the code in 7bff33ce3ce15a29a8279f293f56102b04f56f47. Thanks for the thorough review!